src/compose/service.coffee

```diff
@@ -273,7 +271,7 @@ module.exports = class Service
 			@extendLabels(opts.imageInfo)
 			@extendAndSanitiseVolumes(opts.imageInfo)
 			@extendAndSanitiseExposedPorts(opts.imageInfo)
-			{ @exposedPorts, @portBindings } = @getPortsAndPortBindings()
+			@portMappings = @getPortsAndPortBindings()
```

Comment:
You should also create the portMappings when there is a `@containerId` for the comparison to work (see my comment below on avoiding mutation) - but in that case you'll probably need the inverse function to generatePortBindings, that takes exposedPorts and portBindings (which come from the docker daemon) and produces portMappings